packages/theming/src/utils/componentStyles.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import { DataAttributes, DefaultTheme, ThemeProps } from 'styled-components';
+
+/**
+ * CSS for component customizations based on `theme.components[componentId]`.
+ *
+ * @param {Object} props.theme Provides `components` object use to resolve the given component ID
+ * @param {String} [props.componentId] Specifies the lookup id for * `theme.components` styles.
+ *  The ID will be inferred from the `'data-garden-id'` attribute if not provided.
+ *
+ * @returns component CSS styles
+ */
+export const componentStyles = (props: ThemeProps<DefaultTheme> & { componentId?: string }) => {
```

Comment:
ThemeProps was dropped in `styled-component` v6. To reduce the refactoring scope within our internal `styled` components, I [added back](https://github.com/zendeskgarden/react-components/blame/main/utils/build/styled.d.ts#L16-L18) `ThemeProps` to the `styled-components` namespace. However, `ThemeProps` won't be available for users on `styled-components` v6.

Here's the built output for `componentStyles.d.ts` showing the missing type with v6. 

![Screenshot 2024-12-06 at 11 00 19 AM](https://github.com/user-attachments/assets/4fdf94fd-24d4-429c-aaba-3f29bfe54ecb)

It's probably best to explicitly define the `theme` prop.

```suggestion
export const componentStyles = (props: { componentId?: string, theme: DefaultTheme }) => {
```